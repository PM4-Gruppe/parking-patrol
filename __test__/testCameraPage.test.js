import React from 'react';
import renderer from 'react-test-renderer';
import Camera from '../src/pages/camera';
import { useRouter } from 'next/router';
import { mocked } from 'jest-mock'

jest.mock('next/router', () => ({
    useRouter: jest.fn(),
}));

describe('Camera', () => {
    beforeEach(() => {
        mocked(useRouter).mockReturnValue({
            back: jest.fn(),
        })
    });

    afterEach(() => {
        jest.resetAllMocks()
    });

    it('should match snapshot', () => {
        const tree = renderer.create(<Camera />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});