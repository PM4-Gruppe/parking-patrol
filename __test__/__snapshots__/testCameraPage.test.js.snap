// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera should match snapshot 1`] = `
<div
  className="flex flex-col bg-gradient-to-t from-neutral-800 to-neutral-700 min-h-screen"
>
  <div
    className="flex flex-col items-center justify-center"
  >
    <div>
      <input
        accept="image/*"
        className="displayflex flex-col items-center justify-center w-full p-2 file:bg-green-700 file:hover:bg-green-800 file:text-white text-white file:rounded-lg file:border-transparent my-2"
        onChange={[Function]}
        title="file"
        type="file"
      />
    </div>
    <div
      className="box mb-4 "
    >
      <input
        className="w-full md:w-96 p-0 border-transparent rounded-lg bg-transparent text-white placeholder-neutral-400"
        onChange={[Function]}
        placeholder="Autonummer"
        type="text"
        value=""
      />
      <hr
        className="w-full my-1 text-left ml-0"
      />
      <p
        className="text-white text-xs"
      >
        Information
      </p>
    </div>
    <div
      className="box mb-4 "
    >
      <input
        className="w-full md:w-96 p-0 border-transparent rounded-lg bg-transparent text-white placeholder-neutral-400"
        onChange={[Function]}
        placeholder="Marke"
        type="text"
        value=""
      />
      <hr
        className="w-full my-1 text-left ml-0"
      />
      <p
        className="text-white text-xs"
      >
        Bitte geben Sie eine Marke ein.
      </p>
    </div>
    <div
      className="box mb-4 "
    >
      <input
        className="w-full md:w-96 p-0 border-transparent rounded-lg bg-transparent text-white placeholder-neutral-400"
        onChange={[Function]}
        placeholder="Modell"
        type="text"
        value=""
      />
      <hr
        className="w-full my-1 text-left ml-0"
      />
      <p
        className="text-white text-xs"
      >
        Bitte geben Sie ein Model ein.
      </p>
    </div>
    <div
      className="flex flex-grow-0 justify-between w-1/2"
    >
      <button
        className="p-2 bg-green-700 hover:bg-green-800 text-white rounded-lg my-2"
        onClick={[Function]}
        type="button"
      >
        Zurück
      </button>
      <button
        className="p-2 bg-green-700 hover:bg-green-800 text-white rounded-lg my-2"
        onClick={[Function]}
        type="button"
      >
        Prüfen
      </button>
    </div>
  </div>
</div>
`;
